openwisp_controller/config/tests/test_controller.py

```diff
@@ -430,6 +431,47 @@ def test_register(self, **kwargs):
     def test_register_with_management_ip(self):
         self.test_register(management_ip='10.0.0.2')
 
+    def test_register_exceeds_org_device_limit(self):
+        org = self._get_org()
+        org.config_limits.device_limit = 1
+        org.config_limits.save()
+        self._create_device(organization=org)
+        self.assertEqual(Device.objects.count(), 1)
+        org = self._get_org()
+        response = self.client.post(
+            self.register_url,
+            data={
+                'hardware_id': '12345',
+                'secret': TEST_ORG_SHARED_SECRET,
+                'name': 'second-device',
+                'mac_address': '11:22:33:44:55:66',
+                'backend': 'netjsonconfig.OpenWrt',
+            },
+        )
+        error_dict = json.loads(response.content.decode())
+        self.assertEqual(
+            error_dict['__all__'][0],
+            f'Device registration limit reached for {org} organization',
```

Comment:
Did you test this with a device? Is this error shown in the logread output?

Reply:
It appears like this in the logread output 

```text
root@pandafy-archer-c50:~# /etc/init.d/openwisp_config restart; logread -f
openwisp: OpenWISP config agent stopping
openwisp: OpenWISP config agent started
Mon Oct 17 15:28:48 2022 daemon.info openwisp: Registering device...
Mon Oct 17 15:28:48 2022 daemon.err openwisp: Registration failed! {     "__all__": [         "The maximum amount of allowed devices has been reached for organization default."     ] }
```

Reply:
@nemesisdesign the code sends the error dict in the response (also present in the current master branch)
 
 https://github.com/openwisp/openwisp-controller/blob/fde525f64f835eb374a1b76bf15f23540649aa99/openwisp_controller/config/controller/views.py#L215-L226